pinot-core/src/main/java/org/apache/pinot/core/indexsegment/generator/SegmentGeneratorConfig.java

```diff
@@ -112,7 +112,7 @@
   private String _simpleDateFormat = null;
   // Use on-heap or off-heap memory to generate index (currently only affect inverted index and star-tree v2)
   private boolean _onHeap = false;
-  private boolean _checkTimeColumnValidityDuringGeneration = true;
+  private boolean _skipTimeValueCheck = false;
```

Comment:
If the default is false, how will we catch invalid time values during segment generation?  We will have to push this to users (outside Pinot code) who are using SegmentGeneratorConfig. Earlier, the default was true but the check was disabled for realtime and some unit tests where fixing the invalid time values required rewriting a set of Avro files that were used by tests. Still in some other tests, the check was enabled because those tests were doing data generation programmatically (and so easier to control generation of valid time values) as opposed to Avro test files.

I feel the default should be true.

Reply:
+1 for default being true.
The reason this check was added was because the time column check was done when the segment was pushed, and the segment push failed. The idea was to flag the error earlier since the segment is invalid and can never be pushed anyway

Reply:
@siddharthteotia @mcvsubbu What I am saying is that for a config, the default value for boolean should be false (easier to use because user usually assumes that), and that is the reason why I changed the variable from `_checkTimeColumnValidityDuringGeneration` to `_skipTimeValueCheck`. I agree we should check time value by default.

Reply:
Got it. Thanks @Jackie-Jiang